pyscf/tools/molden.py

```diff
@@ -239,8 +239,16 @@ def read_one_bas(lsym, nb, fac=1):
         elif dat[0].upper() in 'SPDFGHIJ':
             basis[symb].append(read_one_bas(*dat))
 
-    mol.basis = envs['basis'] = basis
     mol.atom = [atoms[i] for i in atom_seq]
+    uniq_atoms = {a[0] for a in mol.atom}
+
+    # To avoid the mol.build() sort the basis, disable mol.basis and set the
+    # internal data _basis directly. It is a workaround to solve issue #1961.
+    # Mole.decontract_basis function should be rewritten to support
+    # discontinuous bases that have the same angular momentum.
+    mol.basis = {}
+    _basis = gto.mole._parse_default_basis(basis, uniq_atoms)
```

Comment:
I guess this somewhat limits what one can do with objects retrieved from molden files, depending on how many functions depend on the sorted basis functions and how common these non standard sortings are.

Reply:
The case of un-sorted basis was considered here. Calling mol.build() will enforce a sort in the function. However, assigning the ._basis attribute of the Mole object directly will skip the sorting. Probably a better solution is to use an input parameter in Mole object to control whether to sort the basis. Introducing the new parameters may require other updates in Mole class. To solve the issue 1961, we can use this temporary solution for the moment